atomicfu/src/jvmMain/kotlin/kotlinx/atomicfu/parking/JvmParkingDelegator.kt

```diff
@@ -0,0 +1,43 @@
+package kotlinx.atomicfu.parking
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.locks.LockSupport
+import kotlin.time.DurationUnit
+import kotlin.time.TimeSource.Monotonic
+
+// Only for testing purposes
+internal class JvmParkingDelegator: ParkingDelegator {
+    private var thread: Thread? = null
+    private val atomicLong: AtomicLong = AtomicLong(0L)
+    
+    override fun createRef(): Long {
+        thread = Thread.currentThread()
+        atomicLong.set(0L)
```

Comment:
It seems to me like the atomic long here is a protection against spurious wake-ups. Is that it? I don't understand the utility of this safety mechanism. Code using thread parking is often already structured in a way that takes spurious wake-ups into account.

If we do remove the safety mechanism, however, it means that `JvmParkingDelegator` can be stateless and use `Thread` as the reference that's passed around. *That*, in turn, means that there's no need to create a parking delegator instance per thread parker.

Reply:
I have made the it stateless. However I do still check the spurious wakeups to have similar semantics to the native implementation. Why native needs this check has to do with memory deallocation (which is discussed further there)